File: lab5-demos/retrofit-demo-java/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceRepository.java
package com.smd.retrofitdemo;

import android.util.Log;
import com.smd.retrofitdemo.network.PeopleInSpace;
import com.smd.retrofitdemo.network.PeopleInSpaceProvider;
import com.smd.retrofitdemo.network.PeopleInSpaceService;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Provides data describing the current people in space. It obtains it using the PeopleInSpaceService.
 */
class PeopleInSpaceRepository {

    private static final String TAG = PeopleInSpaceRepository.class.getName();
    private final MutableLiveData<PeopleInSpace> responseLiveData = new MutableLiveData<>();
    private final PeopleInSpaceService peopleInSpaceService = PeopleInSpaceProvider.getApiService();

    public void getPeopleInSpace() {
        peopleInSpaceService.getPeopleInSpace().enqueue(new Callback<PeopleInSpace>() {
            @Override
            public void onResponse(Call<PeopleInSpace> call, Response<PeopleInSpace> response) {
                if (response.body() != null) {
                    Log.d(TAG, "people in space response received " + response.body());
                    responseLiveData.postValue(response.body());
                }
            }

            @Override
            public void onFailure(Call<PeopleInSpace> call, Throwable throwable) {
                responseLiveData.postValue(null);
            }
        });
    }

    public LiveData<PeopleInSpace> getPeopleInSpaceResponseLiveData() {
        return responseLiveData;
    }
}

File: lab5-demos/retrofit-demo-kotlin/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceRepository.kt
package com.smd.retrofitdemo

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.smd.retrofitdemo.network.PeopleInSpace
import com.smd.retrofitdemo.network.PeopleInSpaceProvider.apiService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import timber.log.Timber

/**
 * Provides data describing the current people in space. It obtains it using the PeopleInSpaceService.
 */
internal class PeopleInSpaceRepository {
    private val responseLiveData = MutableLiveData<PeopleInSpace?>()
    private val peopleInSpaceService = apiService
    val peopleInSpace: Unit
        get() {
            peopleInSpaceService.peopleInSpace().enqueue(object : Callback<PeopleInSpace?> {
                override fun onResponse(
                    call: Call<PeopleInSpace?>,
                    response: Response<PeopleInSpace?>
                ) {
                    if (response.body() != null) {
                        Timber.d("people in space response received ${response.body()}")
                        responseLiveData.postValue(response.body())
                    }
                }

                override fun onFailure(call: Call<PeopleInSpace?>, throwable: Throwable) {
                    responseLiveData.postValue(null)
                }
            })
        }

    val peopleInSpaceResponseLiveData: LiveData<PeopleInSpace?>
        get() = responseLiveData
}